discojs/discojs-core/src/async_informant.ts

```diff
@@ -1,85 +1,64 @@
-import { AsyncBuffer } from './async_buffer'
+import { AggregatorBase } from './aggregator'
 
 export class AsyncInformant<T> {
-  private round = 0
-  private currentNumberOfParticipants = 0
-  private totalNumberOfParticipants = 0
-  private averageNumberOfParticipants = 0
+  private _round = 0
+  private _currentNumberOfParticipants = 0
+  private _totalNumberOfParticipants = 0
+  private _averageNumberOfParticipants = 0
 
   constructor (
-    private readonly asyncBuffer: AsyncBuffer<T>
-  ) {
-    this.asyncBuffer.registerObserver(this)
-  }
-
-  // Update functions
+    private readonly aggregator: AggregatorBase<T>
+  ) {}
 
   update (): void {
-    // DEBUG
-    console.log('Before update')
+    console.debug('before:')
     this.printAllInfos()
-
-    this.updateRound()
-    this.updateNumberOfParticipants()
-
-    // DEBUG
-    console.log('After update')
+    if (this.round === 0 || this.round < this.aggregator.round) {
+      this._round = this.aggregator.round
+      this._currentNumberOfParticipants = this.aggregator.size
+      this._averageNumberOfParticipants = this.totalNumberOfParticipants / this.round
+      this._totalNumberOfParticipants += this.currentNumberOfParticipants
+    } else {
+      this._round = this.aggregator.round
```

Comment:
this is here because the `update` method is called once per communication round, meaning possibly multiple times during a single training round, but the stats are over training rounds

but yes the code is unclear 😁